api/elastic.py

```diff
@@ -234,30 +234,37 @@ def titleQuery(self, titleText):
 
     #Query for the identifier authority followed up with the identifier value
     def identifierQuery(self, authorityList, authIdentText):
-        self.searchedFields.extend(['identifiers', 'editions.identifiers'])
+        self.searchedFields.extend(['identifiers.identifier', 'editions.identifiers.identifier',
+                                    'identifiers.authority', 'editions.identifiers.authority'])
 
         authIdentList = authIdentText.split(': ')
 
         #When user only types in the identifier in the search bar
         if len(authIdentList) < 2:
 
-            return Q('bool', should=[
-            Q(
-                'query_string',
-                query=authIdentText,
-                fields=['identifiers.*'],
-                default_operator='and'
-            ),
-            Q(
-                'nested',
-                path='editions',
-                query=Q(
-                    'query_string',
-                    query=authIdentText,
-                    fields=['editions.identifiers.*'],
+            identifier = authIdentText
+
+            workIdentQuery = Q('bool', must=[
+                Q(
+                    'term',
+                    query=identifier,
+                    fields=['identifiers.identifier'],
                     default_operator='and'
                 )
-            )
+            ])
+
+            editionIdentQuery = Q('bool', must=[
+                Q(
+                    'term',
+                    query=identifier,
+                    fields=['editions.identifiers.identifier'],
+                    default_operator='and'
```

Comment:
with the `term` query we can simplify this because we don't need these specific keys, actually just need field and query: `'term', **{'editions.identifiers.identifier': identifier} (need the dictionary unpacking because we need the string as the key)